pydatastructs/trees/heaps.py

```diff
@@ -0,0 +1,223 @@
+from pydatastructs.utils.misc_util import _check_type
+
+NoneType = type(None)
+
+__all__ = [
+        'BinaryHeap'
+]
+
+class BinaryHeap():
+    """
+    Respresentation of BinaryHeap data structure
+
+    Parameters
+    ==========
+
+    array : list
+        Optional, by default 'None'
+        List of initial elements in Heap
+        
+    _type : str
+        Type of Heap.
+        Takes 'min' or 'max'
+        By default 'min'
+    
+    References
+    ==========
+
+    .. [1] https://en.m.wikipedia.org/wiki/Binary_heap
+    
+    """
+    def __new__(cls, array=None, _type="min"):
+        if _type=="min":
+            return MinHeap(array)
+        elif _type=="max":
+            return MaxHeap(array)
+        else:
+            raise NotImplementedError("%s hasn't been implemented yet."%(_type))
+    
+    def insert(self, *args, **kwargs):
+        """
+        Insert a new element to the Heap according to heap property.
+
+        Parameters
+        ==========
+
+        new_key: float
+            A real number to insert into the Heap.
+        
+        """
+        raise NotImplementedError(
+            "This is an abstract method.")
+
+    def extract(self):
+        """
+        Extract root element of the Heap.
+
+        Returns
+        ==========
+        
+        element_to_be_extracted : float

```

Comment:
It looks like it doesn't need the arguments.

Reply:
Should i add 'Heap' in the parameters?

Reply:
I don't think it's needed.